Codecademy/gamut
# chore(HeaderTab and HeaderContainer)!: remove old unused components

### Overview


Removes old and unused `HeaderTab` and `HeaderContainer` components.


### PR Checklist

- [x] Related to JIRA ticket: [WEB-1847]

#### Testing Instructions


See testing instructions in https://github.com/codecademy-engineering/Codecademy/pull/31581

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31581)  | [Monolith Env](https://pr-31581-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2256)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2256)   |
